assets/section-main-product.css

```diff
@@ -505,6 +510,10 @@ a.product__text {
   .product__media-list .product__media-item {
     width: calc(100% - 3rem - var(--grid-mobile-horizontal-spacing));
   }
+
+  .product--mobile-columns .product__media-item {
+    width: calc(50% - 1.5rem - var(--grid-mobile-horizontal-spacing));
+  }
```

Comment:
Thanks Ludo! One of the other, very many things to additionally account / test for 😅 but nonetheless, this is a great catch! 👍🏼

Reply:
https://github.com/Shopify/dawn/pull/1766/commits/6d11ca179ebfd2bcc61e6b6a413cf6bb0560c19a

I added another condition when using mobile columns so that the media count is the same regardless of `section.settings.hide_variants` / "Hide other variants’ media after selecting a variant". Then I added additional logic to `slider-buttons`. I tested many products with different setting combinations and if there's a better way to implement what I've added above please feel free to suggest them 👍🏼 

To check products with 3 and 2 images on mobile, navigate to Shop all and view the following products:

- Gen Crop Pants (3 images)
- Otti Summer Boot - Beige